deployments/databricks/deploy_collector.sh

```diff
@@ -0,0 +1,170 @@
+# limitations under the License.
+
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+# This script is used to deploy the Splunk distribution of the OpenTelemetry Collector
+# on the current node of a Databricks cluster. Through UI configuration the script will
+# be distributed and run on every node of the cluster.
+
+# Required Variables:
+# - SPLUNK_ACCESS_TOKEN: Splunk o11y access token for sending data to Splunk o11y backend.
+# - DATABRICKS_CLUSTER_HOSTNAME: Hostname of the Databricks compute resource. Use the Server
+#   Hostname described here:
+#   https://docs.databricks.com/en/integrations/compute-details.html
+# - DATABRICKS_ACCESS_TOKEN: Databricks personal access token (PAT) used to connect to the Apache Spark API.
+#   Directions for creating a PAT: https://docs.databricks.com/en/dev-tools/auth/pat.html
+
+# Optional Variables:
+# - OTEL_VERSION: Version of the Splunk OpenTelemetry Collector to deploy as a part of this release.
+#   Default: "latest". Valid version must be >=0.120.0.
+# - SCRIPT_DIR: Installation path for the Collector and its config
+# - SPLUNK_REALM: Splunk o11y realm to send data to. Default: us0
+
+DATABRICKS_CLUSTER_HOSTNAME=${DATABRICKS_CLUSTER_HOSTNAME:-}
+SPLUNK_ACCESS_TOKEN=${SPLUNK_ACCESS_TOKEN:-}
+DATABRICKS_ACCESS_TOKEN=${DATABRICKS_ACCESS_TOKEN:-}
+
+if [ -z "${DATABRICKS_CLUSTER_HOSTNAME}" ]; then
+  echo "environment variable 'DATABRICKS_CLUSTER_HOSTNAME' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SPLUNK_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'SPLUNK_ACCESS_TOKEN' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SDATABRICKS_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'DATABRICKS_ACCESS_TOKEN' must be set, exiting."
+fi
+
+
+OTEL_VERSION=${OTEL_VERSION:-latest}
+OS="linux_amd64"
+OTEL_BINARY_NAME="splunk_otel_collector"
+OTEL_DOWNLOAD_BASE_URL="https://github.com/signalfx/splunk-otel-collector/releases"
+OTEL_API_URL="https://api.github.com/repos/signalfx/splunk-otel-collector/releases/latest"
+SCRIPT_DIR=${SCRIPT_DIR:-/tmp/collector_download}
+CONFIG_FILENAME="config.yaml"
+OTEL_BINARY_FILE="$SCRIPT_DIR/$OTEL_BINARY_NAME"
+CONFIG_FILE="$SCRIPT_DIR/$CONFIG_FILENAME"
+SERVICE_PATH="/etc/systemd/system/"
+SERVICE_FILE="$SERVICE_PATH/$OTEL_BINARY_NAME.service"
+
+if [ $OTEL_VERSION = "latest" ]; then
+        OTEL_VERSION=$(curl --silent "$OTEL_API_URL" |    # Get latest Collector release from GitHub api
+            grep '"tag_name":' |                          # Get tag name line
+            sed -E 's/.*"([^"]+)".*/\1/')                 # Pluck latest release version
+        if [ -z "$OTEL_VERSION" ]; then
+            echo "Failed to get tag_name for latest release from $OTEL_VERSION/latest" >&2
+            exit 1
+        fi
+fi
+
+OTEL_BINARY_DOWNLOAD_URL=${OTEL_DOWNLOAD_BASE_URL}/download/${OTEL_VERSION}/otelcol_${OS}
+
+# Download Splunk's distribution of the OpenTelemetry Collector
+mkdir -p "$SCRIPT_DIR"
+wget -nv -O "$OTEL_BINARY_FILE" $OTEL_BINARY_DOWNLOAD_URL
```

Comment:
```suggestion
wget -nv -O "$OTEL_BINARY_FILE" "$OTEL_BINARY_DOWNLOAD_URL" || { echo "Failed to download $OTEL_BINARY_DOWNLOAD_URL"; exit 1; }
```

Should we use only one of either wget or curl (curl is used to get some info earlier in the script)? Not a big deal especially if we are certain that a databricks host will have both.

Reply:
I've moved to use `curl` for both commands. Both `wget` and `curl` should be installed, but consistency can't hurt.